python-threatexchange/threatexchange/exchanges/impl/techagainstterrorism_api.py

```diff
@@ -90,27 +99,17 @@ def get_client(self) -> api.TATHashListAPI:
     def fetch_iter(
         self,
         _supported_signal_types: t.Sequence[t.Type[SignalType]],
-        checkpoint: t.Optional[state.TFetchCheckpoint],
+        checkpoint: t.Optional[TATCheckpoint],
     ) -> t.Iterator[_TypedDelta]:
 
         client = self.get_client()
-        result = client.get_hash_list()
-
-        translated = (_get_delta_mapping(entry) for entry in result)
 
-        yield state.FetchDelta(
-            dict(t for t in translated if t[0][0]),
-            checkpoint=state.NoCheckpointing(),
-        )
-
-
-def _is_compatible_signal_type(record: t.Dict[str, str]) -> bool:
-    compatible_video_types = ["mov", "m4v", "mp4", "webm"]
-
-    if record["algorithm"] == "MD5":
-        return record["file_type"] in compatible_video_types
-
-    return record["algorithm"] == "PDQ"
+        for result in client.fetch_hashes_iter(checkpoint.checkpoint):
+            translated = (_get_delta_mapping(r) for r in result["results"])
+            yield state.FetchDelta(
+                dict(t for t in translated if t[0][0]),
+                TATCheckpoint.from_tat_fetch(result),
```

Comment:
I suspect your bug is either L107 or L111

blocking question: if your API returns zero hashes:
1. Does it yield a response?
2. If so, what does the response checkpoint show as?

(Expected - if the checkpoint is not advanced, this method should either yield nothing, or should yield zero records with a valid checkpoint of where to call next)

Reply:
Thanks David,

Looking at our API it will return the following if there are no hashes left.
```
{
    "count": 19636,
    "next": null,
    "previous": "http://dev.terrorismanalytics.org/hash-list/vafter=1730204429.203484%2C28022&limit=1000&offset=19000&order=asc",
    "checkpoint": null,
    "results": []
}
```

In this case inside `techagainsterroism_api.py` I am checking for this and returning early.

```
for result in client.fetch_hashes_iter(start_time):

    if result.next is None:
        return

    translated = (_get_delta_mapping(r) for r in result.results)
    yield state.FetchDelta(
        dict(t for t in translated if t[0][0]),
        TATCheckpoint.from_tat_fetch(result),
    )
```

This seems to have resolved the issue and now the last checkpoint to get set will remain in place.

Reply:
Suggest that when there are no hashes left, your API should either:
1. [Recommended] Return the checkpoint corresponding to the last record (latest checkpoint)
2. Return the checkpoint passed by the user.